v2/internal/genericarmclient/generic_client.go

```diff
@@ -115,16 +116,25 @@ func (client *GenericClient) BeginCreateOrUpdateByID(ctx context.Context, resour
 
 func (client *GenericClient) createOrUpdateByID(
 	ctx context.Context,
-	resourceID string,
+	resourceID,
+	resourceName,
```

Comment:
Do you mean this to be the resource type `ResourceGroup`, or the name of the resource? What you have here is the name of the resource, like "myFoo", which I don't think you want?

Reply:
It's also possible to parse this from the `resourceID` so you might not need a separate parameter.

Reply:
True, `kind` would make more sense here?

Reply:
can surely parse `type` from resourceID but would that be a good idea? Would resourceID format be changing from resource to resource ?

Reply:
There's a defined format and a parse method. See [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/arm/resource_identifier.go#L64). We use this as `arm.ParseResourceID` in a few places already. I _think_ the string you want is:

```
parsedID, err := arm.ParseResourceID(id)
if err != nil {
    // Handle error
}

fullIdentifier := parsedID.ResourceType.Namespace + "/" + parsedID.ResourceType.Type
```